src/libsync/propagatedownload.cpp

```diff
@@ -484,7 +486,22 @@ void PropagateDownloadFileQNAM::slotGetFinished()
         return;
     }
 
-    downloadFinished();
```

Comment:
You could call downloadFinished in any case if the header did not contains the checksum